### DISCUSSION TITLE
How to ensure a pathname variable is part of my routeTree ?

### DISCUSSION THREAD

**Discussion by seedy** [TanStack/router#1719] [Q&A]
I have a use-case where a global component shows a "go back" link.

I'd like to configure the target of this go back link per Route.

Instead of moving the component from a global layout down to each Route, I'd like to use the `staticData` prop on each Route.

To ensure type-safety, I'd like to somehow - on the type level, function level, or even component level - ensure my pathname is part of my routeTree.

I already tried typing inside `StaticDataRouteOption`, but I'm blocked due to circular references :

```
  interface StaticDataRouteOption {
    i18nKeyTitle?: string;
    backTo?: RoutePaths<typeof routeTree>;
  }
```

![Screenshot 2024-06-06 at 16 34 43](https://github.com/TanStack/router/assets/3367393/35a99e4b-d37f-48e0-a3fb-1ff800c22265)

I was also thinking about a hook which would validate a pathname as being part of my routeTree :

```
const useValidatePathname = () => {
  const {routesByPath} = useRouter();

 return (pathname: string) => routesByPath[pathname] !== undefined
}
```

What do you think ?

**Comment by bisak** [comment]
I have the same question, did you figure this out?

**Comment by mmospanenko** [comment]
Trying to understand the same, very unclear...

**Comment by mmospanenko** [comment]
try this as a type, works for `to`, `from` 
```ts
import type { RegisteredRouter, RouteIds, RoutePaths } from '@tanstack/react-router';

export type RoutePath = RoutePaths<RegisteredRouter['routeTree']>;
export type RouteId = RouteIds<RegisteredRouter['routeTree']>;
```

**Reply by kingmalitha** [reply]
Cannot find name 'RoutePaths'.  From Where did you import RouthPaths ??
version:  @tanstack/react-router 1.114.27

**Reply by mmospanenko** [reply]
> Cannot find name 'RoutePaths'. From Where did you import RouthPaths ?? version: @tanstack/react-router 1.114.27

updated comment

**Reply by kingmalitha** [reply]
seems like they removed it , RoutePaths

![image](https://github.com/user-attachments/assets/90b2a3ff-35a9-4a51-88bf-43f6e897f609)

**Reply by Demnu** [reply]
yea my project has broken because of this.... Don't know what to do :(

**Comment by airtonix** [comment]
Your `routeTree.gen.ts` should have a generated type `FileRoutesByPath`.

So maybe this works ? 

`app/router.ts`
```ts
declare module "@tanstack/react-router" {
  interface Register {
    router: ReturnType<typeof createRouter>;
  }
  interface StaticDataRouteOption {
    i18nKeyTitle?: string;
    backTo?: keyof FileRoutesByPath;
  }
}
```

![image](https://github.com/user-attachments/assets/6c82a54c-1736-4bd4-aa2e-4d39bc598443)


![image](https://github.com/user-attachments/assets/41d387c0-72bb-42d9-b1a8-dca644bdec01)

You could also use `keyof FileRoutesById`, `FileRouteTypes['id']`
